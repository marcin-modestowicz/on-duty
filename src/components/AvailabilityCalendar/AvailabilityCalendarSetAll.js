// @flow
import React, { Component } from "react";
import { observable } from "mobx";
import { observer } from "mobx-react";
import classnames from "classnames";
import {
  type AvailabilityStatus,
  AVAILABILITY_STATUSES,
  AVAILABILITY_STATUSES_NAMES
} from "../../models/Availability";
import styles from "./AvailabilityCalendarSetAll.scss";

type Props = {
  onClick: AvailabilityStatus => void
};

class AvailabilityCalendarSetAll extends Component<Props> {
  handleToggleAll = (status: AvailabilityStatus) => {
    return () => {
      this.props.onClick(status);
    };
  };

  render() {
    return (
      <div className={styles.root}>
        <div className={styles.label}>Set all:</div>
        {Object.keys(AVAILABILITY_STATUSES).map(statusKey => {
          const status = AVAILABILITY_STATUSES[statusKey];
          const statusName = AVAILABILITY_STATUSES_NAMES[status];

          return (
            <div
              key={status}
              role="button"
              className={classnames(
                styles.button,
                styles[statusName.toLowerCase()]
              )}
              title={`Click to set all to ${statusName}`}
              onClick={this.handleToggleAll(status)}
            >
              {statusName}
            </div>
          );
        })}
      </div>
    );
  }
}

export default AvailabilityCalendarSetAll;
