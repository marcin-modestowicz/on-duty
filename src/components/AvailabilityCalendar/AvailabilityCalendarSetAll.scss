@import '../../styles/colors.scss';

.root {
  display: flex;
  flex-wrap: wrap;

  .label {
    box-sizing: border-box;
    width: 5em;
    height: 2.5em;
    padding: .75em;
    margin: .25em;
  }
}

.button {
  width: 5em;
  margin: .25em;
}