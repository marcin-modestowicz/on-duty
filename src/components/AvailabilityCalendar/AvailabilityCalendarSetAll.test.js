// @flow
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AvailabilityCalendarSetAll from "./AvailabilityCalendarSetAll";

describe("AvailabilityCalendarSetAll component", () => {
  let availabilityCaledarSetAll;

  beforeAll(() => {
    availabilityCaledarSetAll = shallow(
      <AvailabilityCalendarSetAll onClick={jest.fn()} />
    );
  });

  test("should render properly", () => {
    expect(toJson(availabilityCaledarSetAll)).toMatchSnapshot();
  });
});
