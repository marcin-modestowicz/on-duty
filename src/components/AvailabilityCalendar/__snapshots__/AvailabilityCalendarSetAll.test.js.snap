// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityCalendarSetAll component should render properly 1`] = `
<div
  className="root"
>
  <div
    className="label"
  >
    Set all:
  </div>
  <Button
    className="button"
    color="yellow"
    key="0"
    onClick={[Function]}
    title="Click to set all to Free"
  >
    Free
  </Button>
  <Button
    className="button"
    color="green"
    key="1"
    onClick={[Function]}
    title="Click to set all to Keen"
  >
    Keen
  </Button>
  <Button
    className="button"
    color="red"
    key="-1"
    onClick={[Function]}
    title="Click to set all to Busy"
  >
    Busy
  </Button>
</div>
`;
